File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    l = list(polynomial.replace(" ",'').split('+'))
    x_sum = 0
    n_sum = 0
    for i in l:
        if i[-1] == 'x':
            if len(i) ==1:
                x_sum += 1
            else:
                x_sum += int(i[0:len(i)-1])
        else:
            n_sum += int(i)
    if x_sum == 0:
        return f'{n_sum}'
    elif x_sum != 0 and n_sum == 0:
        if x_sum == 1:
            return 'x'
        else:
            return f'{x_sum}x'
    else:
        if x_sum == 1:
            return f'x + {n_sum}'
        else:
            return f'{x_sum}x + {n_sum}'

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.java
class Solution {
    public String solution(String polynomial) {
        String[] terms = polynomial.split(" \\+ ");
        int x_num = 0;
        int num = 0;
        for (String term : terms) {
            if (term.contains("x")) {
                if (term.equals("x")) {x_num++;}
                else {
                    String coefficientStr = term.replace("x", "");
                    x_num += Integer.parseInt(coefficientStr);
                    }
                } else {num += Integer.parseInt(term);}
            }
        StringBuilder answer = new StringBuilder();
        if (x_num != 0) {
            if (x_num == 1) {
                answer.append("x");
            } else {
                answer.append(String.valueOf(x_num)).append("x");
            }
        }
        if (num != 0) {
            if (x_num != 0) {
                answer.append(" + ");
            }
            answer.append(num);
        }
        if (answer.length() == 0) {
            return "0";
        }
        return answer.toString();
    }
}